packages/apidom-ns-asyncapi-3/src/refractor/plugins/replace-empty-element.ts

```diff
@@ -0,0 +1,1236 @@
+    },
+  },
+
+  [ComponentsServerVariablesElement.primaryClass]: {
+    '[key: *]': function key(...args: any[]) {
+      return new ServerVariableElement(...args);
+    },
+  },
+
+  [ComponentsMessagesElement.primaryClass]: {
+    '[key: *]': function key(...args: any[]) {
+      return new MessageElement(...args);
+    },
+  },
+
+  [ComponentsSecuritySchemesElement.primaryClass]: {
+    '[key: *]': function key(...args: any[]) {
+      return new SecuritySchemeElement(...args);
+    },
+  },
+
+  [ComponentsParametersElement.primaryClass]: {
+    '[key: *]': function key(...args: any[]) {
+      return new ParameterElement(...args);
+    },
+  },
+
+  [ComponentsCorrelationIDsElement.primaryClass]: {
+    '[key: *]': function key(...args: any[]) {
+      return new CorrelationIDElement(...args);
+    },
+  },
+
+  [ComponentsOperationTraitsElement.primaryClass]: {
+    '[key: *]': function key(...args: any[]) {
+      return new OperationTraitElement(...args);
+    },
+  },
+
+  [ComponentsMessageTraitsElement.primaryClass]: {
+    '[key: *]': function key(...args: any[]) {
+      return new MessageTraitElement(...args);
+    },
+  },
+
+  [ComponentsServerBindingsElement.primaryClass]: {
+    '[key: *]': function key(...args: any[]) {
+      return new ServerBindingsElement(...args);
+    },
+  },
+
+  [ComponentsChannelBindingsElement.primaryClass]: {
+    '[key: *]': function key(...args: any[]) {
+      return new ChannelBindingsElement(...args);
+    },
+  },
+
+  [ComponentsOperationBindingsElement.primaryClass]: {
+    '[key: *]': function key(...args: any[]) {
+      return new OperationBindingsElement(...args);
+    },
+  },
+
+  [ComponentsOperationsElement.primaryClass]: {
+    '[key: *]': function key(...args: any[]) {
+      return new OperationBindingsElement(...args);
+    },
+  },
+
+  [ComponentsMessageBindingsElement.primaryClass]: {
+    '[key: *]': function key(...args: any[]) {
+      return new MessageBindingsElement(...args);
+    },
+  },
+
+  [ComponentsOperationsElement.primaryClass]: {
+    '<*>': function asterisk(...args: any[]) {
+      return new OperationElement(...args);
+    },
+  },
+
+  [ServerVariablesElement.primaryClass]: {
+    '[key: *]': function key(...args: any[]) {
+      return new ServerVariableElement(...args);
+    },
+  },
+
+  [ServerSecurityElement.primaryClass]: {
+    '<*>': function asterisk(...args: any[]) {
+      return new SecurityRequirementElement(...args);
+    },
+  },
+
+  [OperationTraitsElement.primaryClass]: {
+    '<*>': function asterisk(...args: any[]) {
+      return new OperationTraitElement(...args);
+    },
+  },
+
+  [OperationSecurityElement.primaryClass]: {
+    '<*>': function asterisk(...args: any[]) {
+      return new SecurityRequirementElement(...args);
+    },
+  },
+
+  [OperationMessageMapElement.primaryClass]: {
+    oneOf(...args: any[]) {
+      return new OperationMessageElement(...args);
+    },
+  },
+
+  [OperationMessageElement.primaryClass]: {
```

Comment:
I think this one is also a ReferenceElement
```suggestion
  [OperationMessagesElement.primaryClass]: {
     //...
     return new ReferenceElement(...args);
     
  }
```